File: 682-baseball-game/682-baseball-game.py
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        stack = []
        
        for op in ops:
            if op == "C":
                stack.pop()
            elif op == "D":
                curr_score = stack[-1]*2
                stack.append(curr_score)
            elif op == "+":
                curr_score = stack[-1] + stack[-2]
                stack.append(curr_score)
            else:
                stack.append(int(op))
                
        return sum(stack)

File: 682-baseball-game/682-baseball-game.java
class Solution {
    public int calPoints(String[] ops) {
        
        Stack<Integer> stk = new Stack();  
        for (String op: ops) {
            //System.out.println(op);
            if (op.equals("C")) {
                stk.pop();
            } else if (op.equals("D")) {
                //System.out.println(op);
                stk.push(stk.peek() * 2);
            } else if (op.equals("+")) {
                int num1 = stk.pop();
                int num2 = stk.peek();
                //System.out.println(num2);
                stk.push(num1);
                stk.push(num2+num1);
        
            } else {
                //System.out.println(op);
                stk.push(Integer.parseInt(op));
            };
      };
        
    int sum = 0;
    while (stk.size() > 0) {
        int pop= stk.pop();

        sum += pop;
    };
    
    return sum;
        
    };
};